raiden-cli/src/index.ts

```diff
@@ -231,96 +49,66 @@ async function main() {
-  let stopSub: ReturnType<typeof raiden.events$.subscribe>;
-  const raidenStop = () => {
-    if (stopSub) stopSub.unsubscribe();
-    if (server?.listening) server.close();
-    if (!raiden?.started) process.exit(1);
-    log.info('Stopping...');
-    raiden.stop();
-  };
-  process.on('SIGINT', raidenStop);
-  process.on('SIGTERM', raidenStop);
+async function getWallet(privateKeyPath: string, password: string): Promise<Wallet> {
+  const encryptedKey = await fs.readFile(privateKeyPath, 'utf-8');
+  return await Wallet.fromEncryptedJson(encryptedKey, password);
+}
+
+function createLocalStorage(name: string): LocalStorage {
+  const localStorage = new LocalStorage(name);
+  Object.assign(globalThis, { localStorage });
+  return localStorage;
+}
 
-  if (!argv.password) {
-    // argv.password = inquirer.prompt('Private Key Password: ', { hideEchoBack: true });
-    argv.password = (
-      await inquirer.prompt<{ password: string }>([
-        { type: 'password', name: 'password', message: 'Private Key Password:', mask: '*' },
-      ])
-    ).password;
+function shutdown(this: Cli): void {
+  if (this.server?.listening) {
+    this.log.info('Closing server...');
+    this.server.close();
   }
 
-  const wallet = await Wallet.fromEncryptedJson(
-    await fs.readFile(argv.privateKey, 'utf-8'),
-    argv.password,
-  );
-  log = logging.getLogger(`cli:${wallet.address}`);
-  log.info('Address:', wallet.address);
+  if (this.raiden.started) {
+    this.log.info('Stopping raiden...');
+    this.raiden.stop();
+  } else {
+    process.exit(1);
+  }
+}
 
-  const localStorage = new LocalStorage(argv.store);
-  Object.assign(globalThis, { localStorage });
+function registerShutdownHooks(this: Cli): void {
+  process.on('SIGINT', shutdown.bind(this));
+  process.on('SIGTERM', shutdown.bind(this));
+}
 
-  const defaultConfig = {
-    matrixServer: 'https://raidentransport.test001.env.raiden.network',
-    pfs: 'https://pfs.raidentransport.test001.env.raiden.network',
-    pfsSafetyMargin: 1.1,
-    caps: { noDelivery: true, webRTC: true },
-  };
-  raiden = await Raiden.create(argv.ethNode, wallet.privateKey, localStorage, undefined, {
-    ...defaultConfig,
+async function main() {
+  const argv = parseArguments();
+  const password = argv.password ?? (await askUserForPassword());
+  const wallet = await getWallet(argv.privateKey, password);
+  const localStorage = createLocalStorage(argv.store);
+  const log = setupLoglevel();
+  const raiden = await Raiden.create(argv.ethNode, wallet.privateKey, localStorage, undefined, {
+    ...DEFAULT_RAIDEN_CONFIG,
     ...argv.config,
   });
-
-  const init = raidenInit(argv);
-  raiden.start();
-  // if events$ completes, it means raiden shut down
-  stopSub = raiden.events$.subscribe(undefined, raidenStop, raidenStop);
-  try {
-    await init;
-    if (argv.serve) await setupApi(argv.serve);
-  } finally {
-    if (!argv.serve) raidenStop();
-  }
+  const cli = await makeCli(log, raiden, argv.port);
```

Comment:
Looks like `makeCli` is a very small function/module, which actually do only 2 things besides creating the `cli`. Wouldn't be worth it to have these 2 lines (app and server assignment) done in the end of `makeApp` (maybe rename it to `setupApp`), and then have `cli` object creation here, just folowed by a call to `setupApp(cli)`? Looks like `main` already knows about `log` and `raiden` members anyway, so I'd say it'd not deviate much from this. Nitpicking anyway, feel free to ignore this if you think it's worth to keep this separation.

Reply:
To be honest would like to keep it. I think this way the modules are more decoupled from each other. So yes the `cli` module is tiny. My intention was to use it as an accumulator for the REST server. By that it could be used everywhere, providing the arguments as needed. So it is only used by the caller of the command line with its arguments. So the `raiden` and `log` instance are retrieved from the users input. But in other cases this could be created by anyone for any need. So he can simply call `makeCli` and get his own server, that is independent of the other one(s). We may can improve that later? Maybe have a discussion. Shouldn't be much work to change later and from the user perspective nothing will change.